token-client/src/main/java/com/sap/cloud/security/xsuaa/tokenflows/XsuaaTokenFlowRequest.java

```diff
@@ -21,11 +22,12 @@
  */
 class XsuaaTokenFlowRequest implements XSTokenRequest {
 
+	private ClientIdentity clientIdentity;
 	private String clientId;
-	private URI tokenServiceEndpoint;
+	private String clientSecret;
```

Comment:
clientId and clientSecret can be wrapped with clientIdenity

Reply:
@nenaraab you mean remove setters and getters for id and secret from XSTokenRequest interface?

Reply:
here I'm not sure - lets have a look into the code